node/subsystem/src/lib.rs

```diff
@@ -228,14 +228,20 @@ pub trait Subsystem<C: SubsystemContext> {
 /// types of messages. Used for tests or as a placeholder.
 pub struct DummySubsystem;
 
-impl<C: SubsystemContext> Subsystem<C> for DummySubsystem {
+impl<C: SubsystemContext> Subsystem<C> for DummySubsystem
+where
+	C::Message: std::fmt::Debug
+{
 	fn start(self, mut ctx: C) -> SpawnedSubsystem {
 		let future = Box::pin(async move {
 			loop {
 				match ctx.recv().await {
-					Ok(FromOverseer::Signal(OverseerSignal::Conclude)) => return Ok(()),
 					Err(_) => return Ok(()),
-					_ => continue,
+					Ok(FromOverseer::Signal(OverseerSignal::Conclude)) => return Ok(()),
+					Ok(overseer_msg) => {
+						log::debug!("Discarding a message sent from overseer {:?}", overseer_msg);
```

Comment:
if you really want to log messages, you probably want to specify a target (`overseer` or `dummy` maybe?), otherwise you'd have to enable debug for everything, which is a lot of logs